### ISSUE TITLE
Dotnet new for 2.1 sdk should target 2.1

### ISSUE THREAD

**Issue by badersur** [dotnet/cli#7535]
## Steps to reproduce
1. download: https://dotnetcli.blob.core.windows.net/dotnet/Sdk/master/dotnet-sdk-latest-linux-x64.tar.gz on Ubuntu 17 and extract it (Or use [my script](https://github.com/badersur/dotnet-manager) or run `curl https://raw.githubusercontent.com/dotnet/cli/master/scripts/obtain/dotnet-install.sh | bash -s -- -c master`
2. add dotnet to the path: `export PATH="$PATH:<dotnet_location>"`
3. `dotnet new console -o testing` OR `dotnet new mvc -o mvc.net -au Individual -uld`
4. `dotnet run`

## Expected  behavior
The project is running!

## Actual behavior
I get this error:
It was not possible to find any compatible framework version
The specified framework 'Microsoft.NETCore.App', version '2.0.0' was not found.
  - Check application dependencies and target a framework version installed at:
      /
  - Alternatively, install the framework version '2.0.0'.

I've deleted all dotnet files from my home folder: .dotnet, .nuget, etc with no luck! I think the issue started in this release or the previous one.

## Environment data
`dotnet --info` output:
.NET Command Line Tools (2.1.0-preview1-007082)

Product Information:
 Version:            2.1.0-preview1-007082
 Commit SHA-1 hash:  a7eba4d9e8

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  17.04
 OS Platform: Linux
 RID:         linux-x64
 Base Path:   /home/badersur/Downloads/dotnet/sdk/2.1.0-preview1-007082/

Microsoft .NET Core Shared Framework Host

  Version  : 2.1.0-preview2-25616-02
  Build    : 8aa34a95bad481e1b5e4c8286e5cb2d7cb6ed943

I'm wondering if there's a list of old preview releases to re-download!

**Comment by nguerrera** [comment]
If you want to play with these **extremely early daily 2.1 bits out of master**, then you need to change the target framework from netcoreapp2.0 to netcoreapp2.1.

Please use the released 2.0.0 rtm bits from https://www.microsoft.com/net/download/core for production scenarios.

Also, installing both 2.0 and 2.1 sdk side by side will let you use the 2.1 SDK targeting the 2.0 runtime. The 2.1 daily builds contain only the 2.1 runtime. Another option is to install the 2.0 runtime from the link above rather than the full 2.0 sdk.

@mlorbetske, can we get templates inserted to master that target 2.1?

@Petermarcu any update on whether we'll make it so 2.0 apps can just run on 2.1 (i.e. roll forward on minor version if patch only roll forward fails?

**Comment by nguerrera** [comment]
You should pass 2.0 as the channel argument to dotnet-install to get 2.0 bits instead of master which will get you daily 2.1 bits.

**Comment by nguerrera** [comment]
Cc @seancpeters

**Comment by badersur** [comment]
I want to use latest bits out of master and I'm not concerned with production usage yet!

I used `<TargetFramework>netcoreapp2.1</TargetFramework>` on Ubuntu and Windows 7, and got this error:
dotnet restore
```
 Restoring packages for /home/badersur/Coding/Testing/DOTNET2/WillWork/WillWork.csproj...
/home/badersur/Coding/Testing/DOTNET2/WillWork/WillWork.csproj : error NU1102: Unable to find package Microsoft.NETCore.App with version (>= 2.1.0-preview2-25616-02)
/home/badersur/Coding/Testing/DOTNET2/WillWork/WillWork.csproj : error NU1102:   - Found 14 version(s) in nuget.org [ Nearest version: 2.0.0 ]
  Generating MSBuild file /home/badersur/Coding/Testing/DOTNET2/WillWork/obj/WillWork.csproj.nuget.g.props.
  Generating MSBuild file /home/badersur/Coding/Testing/DOTNET2/WillWork/obj/WillWork.csproj.nuget.g.targets.
  Restore failed in 172.39 ms for /home/badersur/Coding/Testing/DOTNET2/WillWork/WillWork.csproj.
```

On windows, I have 2.0 installed!

**Comment by nguerrera** [comment]
@badersur This is because daily builds (again, bleeding edge here) of the .NET Core shared framework are not published to nuget.org. You'll need to add a feed via NuGet.config to pick up the corresponding daily builds of the framework packages from myget.org.

```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
     <add key="dotnet-core" value="https://dotnet.myget.org/F/dotnet-core/api/v3/index.json"  />
  </packageSources>
</configuration>
```

See https://docs.microsoft.com/en-us/nuget/consume-packages/configuring-nuget-behavior

**Comment by badersur** [comment]
I think including this feed to preview templates will be awesome!

Thank you @nguerrera for your replies; I'm able to run my apps by installing 2.0 and 2.1 SDKs (even though I don't like the fact I'm installing 2 versions!) :+1:

**Comment by nguerrera** [comment]
Fixed by #7554
